lib/routes/api/accounts.js

```diff
@@ -213,15 +217,48 @@ async function validateCreateCall(logger, sid, req) {
   }
   else {
     delete obj.application_sid;
+    // Fetch Speech provider credential From DB
+    let speech_creds = await SpeechCredential.retrieveAll(req.user.account_sid);
+    if (!speech_creds || speech_creds.length === 0) {
+      // try to fetch speech cred from service provider
+      const {lookupAccountBySid} = req.app.locals;
+      const account = await lookupAccountBySid(req.user.account_sid);
+      speech_creds = await SpeechCredential.retrieveAllForSP(account.service_provider_sid);
+    }
 
-    // TODO: these should be retrieved from account, using account_sid if provided
-    Object.assign(obj, {
-      speech_synthesis_vendor: 'google',
-      speech_synthesis_voice: 'en-US-Wavenet-C',
-      speech_synthesis_language: 'en-US',
-      speech_recognizer_vendor: 'google',
-      speech_recognizer_language: 'en-US'
+    if (!speech_creds || speech_creds.length === 0)
+      throw new DbErrorBadRequest(`missing speech synthesizer for account ${sid}`);
+    const enabled_tts = [];
+    const enabled_stt = [];
+    speech_creds.forEach((cred) => {
+      if (cred.use_for_tts) enabled_tts.push(cred);
+      if (cred.use_for_stt) enabled_stt.push(cred);
     });
+    if (enabled_tts.length === 0)
+      throw new DbErrorBadRequest(`missing speech synthesizer for account ${sid}`);
+    if (enabled_stt.length === 0)
+      throw new DbErrorBadRequest(`missing speech recognizer for account ${sid}`);
+
+    if (!obj.speech_synthesis_vendor) {
```

Comment:
Let's simplify this whole part.  Let's make it a requirement that in the body of the POST you either
(1) specify an application_sid (in which case we get the speech settings from there), or
(2) you must specify in the body the following speech credentials:
```
speech_synthesis_vendor
speech_synthesis_language
speech_synthesis_voice
speech_recogizer_vendor
speech_recogizer_language
```

This will be a breaking change for anyone that has been relying on the google defaults I have set, but I am ok with that.

Reply:
applied suggestion.